osu.Game/Screens/Play/HUD/UnstableRateCounter.cs

```diff
@@ -0,0 +1,183 @@
+            if (judgement.HitObject.HitWindows is HitWindows.EmptyHitWindows || !judgement.IsHit)
+                return;
+
+            hitOffsets.RemoveAt(hitOffsets.Count - 1);
+            updateUr();
+        }
+
+        private void updateUr()
+        {
+            // At Count = 0, we get NaN, While we are allowing count = 1, it will be 0 since average = offset.
+            if (hitOffsets.Count > 0)
+            {
+                double mean = hitOffsets.Average();
+                double squares = hitOffsets.Select(offset => Math.Pow(offset - mean, 2)).Sum();
+                Current.Value = Math.Sqrt(squares / hitOffsets.Count) * 10;
+                setValid(true);
+            }
+            else
+            {
+                Current.Value = 0;
+                setValid(false);
+            }
+        }
+
+        protected override LocalisableString FormatCount(double count)
+        {
+            return count.ToString("0.00 UR");
+        }
+
+        protected override IHasText CreateText() => new TextComponent
+        {
+            Alpha = alpha_when_invalid,
+        };
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (scoreProcessor == null) return;
+
+            scoreProcessor.NewJudgement -= onJudgementAdded;
+            scoreProcessor.JudgementReverted -= onJudgementReverted;
+        }
+
+        private class TextComponent : CompositeDrawable, IHasText
+        {
+            public LocalisableString Text
+            {
+                get => intPart.Text;
+                set
+                {
+                    //Not too sure about this, is there a better way to go about doing this?
+                    splitValue = value.ToString().Split('.');
+                    intPart.Text = splitValue[0];
+                    decimalPart.Text = splitValue[1];
+                }
+            }
+
+            private string[] splitValue;
+            private readonly OsuSpriteText intPart;
+            private readonly OsuSpriteText decimalPart;
+
+            public TextComponent()
+            {
+                AutoSizeAxes = Axes.Both;
+
+                InternalChild = new FillFlowContainer
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        intPart = new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
+                            Font = OsuFont.Numeric.With(size: 16, fixedWidth: true)
+                        },
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.BottomLeft,
+                            Origin = Anchor.BottomLeft,
+                            Font = OsuFont.Numeric.With(size: 8, fixedWidth: true),
+                            Text = ".",
```

Comment:
assumes that the decimal separator is `.` which is not correct in many cultures.

probably be easier to use `ToLocalisableString()` here, manually chop off the integral part by doing `ur - Math.Truncate(ur)` or similar, and use `.00 UR` as the format specifier passed to `ToLocalisableString()`

Reply:
I've fixed this in the latest commit, which also fixed the above comment at the same point.